src/groups/mqb/mqbauthn/mqbauthn_basicauthenticator.cpp

```diff
@@ -0,0 +1,218 @@
+// Copyright 2016-2025 Bloomberg Finance L.P.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <mqbauthn_basicauthenticator.h>
+
+// MQB
+#include <mqbcfg_messages.h>
+#include <mqbplug_authenticator.h>
+
+// BDE
+#include <bdlt_timeunitratio.h>
+#include <bsl_iostream.h>
+#include <bsl_memory.h>
+#include <bsl_optional.h>
+#include <bsl_string.h>
+#include <bsl_string_view.h>
+#include <bslma_allocator.h>
+#include <bslma_managedptr.h>
+#include <bsls_assert.h>
+
+namespace BloombergLP {
+namespace mqbauthn {
+
+namespace {
+
+const int k_AUTHN_DURATION_SECONDS = 600;
+
+}  // namespace
+
+// -------------------------------
+// class BasicAuthenticationResult
+// -------------------------------
+
+BasicAuthenticationResult::BasicAuthenticationResult(
+    bsl::string_view   principal,
+    bsls::Types::Int64 lifetimeMs,
+    bslma::Allocator*  allocator)
+: d_principal(principal, allocator)
+, d_lifetimeMs(lifetimeMs)
+{
+}
+
+BasicAuthenticationResult::~BasicAuthenticationResult()
+{
+}
+
+bsl::string_view BasicAuthenticationResult::principal() const
+{
+    return d_principal;
+}
+
+const bsl::optional<bsls::Types::Int64>&
+BasicAuthenticationResult::lifetimeMs() const
+{
+    return d_lifetimeMs;
+}
+
+// ------------------------
+// class BasicAuthenticator
+// ------------------------
+
+BasicAuthenticator::BasicAuthenticator(
+    const mqbcfg::AuthenticatorPluginConfig* config,
+    bslma::Allocator*                        allocator)
+: d_allocator_p(allocator)
+, d_credentials(allocator)
+, d_isStarted(false)
+{
+    if (!config) {
+        BALL_LOG_WARN << "No config provided for BasicAuthenticator '"
+                      << name() << "' with mechanism '" << mechanism() << "'.";
```

Comment:
The current log looks like

`MQBAUTHN.BASICAUTHENTICATOR No config provided for BasicAuthenticator 'BasicAuthenticator' with mechanism 'BASIC'`

Can be simpler:

```suggestion
        BALL_LOG_WARN << "No configuration provided, have no credentials";
```

I think `name()` and `mechanism()` should mostly be used by external classes that operate authenticators in a generic way without knowing the concrete authenticator type